docs/en/api/FS-API.md

```diff
@@ -187,23 +212,66 @@ Users can override the default policy class in the
+* [LocalFirstPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/LocalFirstPolicy.html)
+
+  **This is the default policy.**
+
+  > A policy that returns the local worker first, and if the local worker doesn't
+  > exist or have enough availability, will select the nearest worker from the active
+  > workers list with sufficient availability.
+  >
+  > The calculation of which worker gets selected is done for each block write.
+
+* [LocalFirstAvoidEvictionPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/LocalFirstAvoidEvictionPolicy.html)
+
+  This is the same as `LocalFirstPolicy` with the following addition:
 
-    Returns the local worker first, and if it does not have enough capacity of a block,
-    randomly picks a worker from the active workers list. This is the default policy.
+  > A policy that returns the local worker first, and if the local worker doesn't
+  > exist or have enough availability, will select the nearest worker from the active
+  > workers list with sufficient availability.
+  >
+  > The calculation of which worker gets selected is done for each block write.
+  >
+  > The PropertyKey `USER_FILE_WRITE_AVOID_EVICTION_POLICY_RESERVED_BYTES`
+  > (alluxio.user.block.avoid.eviction.policy.reserved.size.bytes)
+  > is used as buffer space on each worker when calculating available space
+  > to store each block.
 
-* **MostAvailableFirstPolicy (alluxio.client.block.policy.MostAvailableFirstPolicy)**
+* [MostAvailableFirstPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/MostAvailableFirstPolicy.html)
 
-    Returns the worker with the most available bytes.
+  > A policy that returns the worker with the most available bytes.
+  >
+  > The policy returns null if no worker is qualified.
 
-* **RoundRobinPolicy (alluxio.client.block.policy.RoundRobinPolicy)**
+* [RoundRobinPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/RoundRobinPolicy.html)
 
-    Chooses the worker for the next block in a round-robin manner and skips workers that do not have
-    enough capacity.
+  > A policy that chooses the worker for the next block in a round-robin manner
+  > and skips workers that do not have enough space.
+  >
+  > The policy returns null if no worker can be found.
 
-* **SpecificHostPolicy (alluxio.client.block.policy.SpecificHostPolicy)**
+* [SpecificHostPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/SpecificHostPolicy.html)
 
-    Returns a worker with the specified host name. This policy cannot be set as default policy.
+  > Always returns a worker with the specified hostname.
+  >
+  > Returns null if no active worker on that hostname found.
+  > Note that since this does not have an empty constructor, this cannot be used as a default policy.
```

Comment:
This is also too much internals. 

If this policy cannot be used, let's not even mention it in the doc.

If this policy works, could you elaborate how to "specify some hosts"?

Reply:
I tested this policy, it does indeed put all blocks onto the worker with the hostname specified by `alluxio.worker.hostname`. Interestingly, when short-circuit + domain socket were enabled they bypassed the `alluxio.user.block.write.location.policy.class=SpecificHostPolicy` setting and just wrote to the local worker instead.

I'm not sure if that's correct behaviour or not; correct for the short-circuit but incorrect for the policy.